earthaccess/auth.py

```diff
@@ -18,10 +18,13 @@ class SessionWithHeaderRedirection(requests.Session):
     """
     Requests removes auth headers if the redirect happens outside the
     original req domain.
-    This is taken from https://wiki.earthdata.nasa.gov/display/EL/How+To+Access+Data+With+Python
     """
 
-    AUTH_HOST = "urs.earthdata.nasa.gov"
+    AUTH_HOSTS: List[str] = [
+        "urs.earthdata.nasa.gov",
+        "cumulus.asf.alaska.edu",
```

Comment:
```suggestion
        "cumulus.asf.alaska.edu",
        "sentinel1.asf.alaska.edu",
```

Sentinel-1 is behined it's own distribution app:
```
curl -L -H "Authorization: Bearer $EDL_TOKEN" https://sentinel1.asf.alaska.edu/GRD_HD/SA/S1A_IW_GRDH_1SDV_20240201T203334_20240201T203408_052370_06552E_14CD.zip -O
```
The ARIA interferrograms (netCDF) are also behind `grfn.asf.alaska.edu` but that distribution app doesn't accept bearer tokens.

Reply:
Do we know anyone with Earthdata Login who can provide a security perspective on this? I feel a bit weird about sending SSO credentials anywhere other than the SSO host!

Reply:
I mean, sending credentials to `datapool.asf.alaska.edu`, which is the domain for the URLs of record in CMR is technically the incorrect part as it doesn't do auth at all (just redirects). `cumulus` and `sentinel1` are the actual auth endpoints

Reply:
I think that's correct, maybe we should fix the OAuth2 behavior for future releases. However this custom domains will still be a problem.

Reply:
>cumulus and sentinel1 are the actual auth endpoints.

What I mean is that, based on my understanding, they should be communicating with the SSO host (EDL) to determine if the user is authenticated/authorized instead of directly receiving credentials from the user. The SSO host acts as an intermediary to reduce credential exposure. Are there other apps on EDL that are taking user credentials? What is EDL's official stance on this practice?

Reply:
@mfisher87 your username and password should only ever be sent to EDL directly, but the bearer token is *intended* to be sent to EDL *apps* (cumulus, tea, etc), and yes, you should be conscious of which domains you're sending your token too. 

Stated here:
![image](https://github.com/nsidc/earthaccess/assets/7882693/c6583c3c-f089-41de-a59b-d5cb86630f1c)

Reply:
Thanks, makes sense! That 2-token limit is so weird.

Maybe we should open an issue to expose a way for end-users to extend the list of approved distribution apps? I assume our list isn't exhaustive or would go out of date.

Reply:
> That 2-token limit is so weird.

It's just like AWS Access keys -- it's really a 1 active key limit, plus space to generate a new one for rotation (you make a new one and then delete the old).

But yeah, it's annoying.